tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,527 @@
+	if err != nil {
+		t.Fatalf("failed to read %v: %v", f, err)
+	}
+	return string(b)
+}
+
+// TestEgressGatewayTls brings up an cluster and will ensure that the TLS origination at
+// egress gateway allows secure communication between the egress gateway and external workload.
+// This test brings up an egress gateway to originate TLS connection. The test will ensure that requests
+// are routed securely through the egress gateway and that the TLS origination happens at the gateway.
+func TestEgressGatewayTls(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.egress.tls.filebased").
+		Run(func(ctx framework.TestContext) {
+			ctx.RequireOrSkip(environment.Kube)
+
+			internalClient, externalServer, appNamespace, _ := setupEcho(t, ctx)
+
+			testCases := map[string]struct {
+				destinationRuleMode string
+				response            []string
+				gateway             bool //  If gateway is true, request is expected to pass through the egress gateway
+				fakeRootCert        bool // If Fake root cert is to be used to verify server's presented certificate
+			}{
+				// Mutual Connection is originated by our DR but server side drops the connection to
+				// only use Simple TLS as it doesn't verify client side cert
+				// TODO: mechanism to enforce mutual TLS(client cert) validation by the server
+				"Mutual TLS origination from egress gateway to https endpoint": {
+					destinationRuleMode: "MUTUAL",
+					response:            []string{response.StatusCodeOK},
+					gateway:             true,
+					fakeRootCert:        false,
+				},
+				"SIMPLE TLS origination from egress gateway to https endpoint": {
+					destinationRuleMode: "SIMPLE",
+					response:            []string{response.StatusCodeOK},
+					gateway:             true,
+					fakeRootCert:        false,
+				},
+				"No TLS origination from egress gateway to https endpoint": {
+					destinationRuleMode: "DISABLE",
+					response:            []string{response.StatusCodeBadRequest},
+					gateway:             false, // 400 response will not contain header
+				},
+				// Egress Gateway can't "originate" ISTIO_MUTUAL to server as the server is outside the mesh and has no
+				// sidecar proxy
+				"ISTIO_MUTUAL TLS origination from egress gateway to https endpoint": {
+					destinationRuleMode: "ISTIO_MUTUAL",
+					response:            []string{response.StatusCodeBadRequest},
+					gateway:             false, // 400 response will not contain header
+					fakeRootCert:        false,
+				},
+				"SIMPLE TLS origination from egress gateway to https endpoint with fake root cert": {
+					destinationRuleMode: "SIMPLE",
+					response:            []string{response.StatusCodeBadRequest},
+					gateway:             false, // 400 response will not contain header
+					fakeRootCert:        true,
+				},
+			}
+
+			for name, tc := range testCases {
+				// Test cases:
+				// 1. Mutual TLS origination from egress gateway to https endpoint:
```

Comment:
nit: why not comment next to the definition of the test?

I almost feel greedy commenting this since having a detailed comment is already so helpful :slightly_smiling_face: